sdk/storage/azblob/zc_leaseClient.go

```diff
@@ -0,0 +1,168 @@
+	if err != nil {
+		return BlobLeaseClient{}, err
+	}
+	con := newConnection(blobURL, cred, options)
+	blobClient := BlobClient{client: &blobClient{con, pathRenameMode}, u: *u}
+	return BlobLeaseClient{BlobClient: blobClient, leaseId: leaseId}, nil
+}
+
+// URL returns the URL endpoint used by the BlobLeaseClient object.
+func (blc BlobLeaseClient) URL() url.URL {
+	return blc.u
+}
+
+// String returns the URL as a string.
+func (blc BlobLeaseClient) String() string {
+	u := blc.URL()
+	return u.String()
+}
+
+// WithPipeline creates a new BlobLeaseClient object identical to the source but with the specified request policy pipeline.
+func (blc BlobLeaseClient) WithPipeline(pipeline azcore.Pipeline) BlobLeaseClient {
+	con := newConnectionWithPipeline(blc.u.String(), pipeline)
+	blobClient := BlobClient{client: &blobClient{con, blc.client.pathRenameMode}, u: blc.u}
+	return BlobLeaseClient{BlobClient: blobClient}
+}
+
+// AcquireLease acquires a lease on the blob for write and delete operations. The lease Duration must be between
+// 15 to 60 seconds, or infinite (-1).
+// For more information, see https://docs.microsoft.com/rest/api/storageservices/lease-blob.
+func (blc *BlobLeaseClient) AcquireLease(ctx context.Context, options *AcquireLeaseBlobOptions) (BlobAcquireLeaseResponse, error) {
+	blobAcquireLeaseOptions, modifiedAccessConditions := options.pointers()
+	resp, err := blc.client.AcquireLease(ctx, blobAcquireLeaseOptions, modifiedAccessConditions)
+	return resp, handleError(err)
+}
+
+// BreakLease breaks the blob's previously-acquired lease (if it exists). Pass the LeaseBreakDefault (-1)
+// constant to break a fixed-Duration lease when it expires or an infinite lease immediately.
+// For more information, see https://docs.microsoft.com/rest/api/storageservices/lease-blob.
+func (blc *BlobLeaseClient) BreakLease(ctx context.Context, options *BreakLeaseBlobOptions) (BlobBreakLeaseResponse, error) {
+	blobBreakLeaseOptions, modifiedAccessConditions := options.pointers()
+	resp, err := blc.client.BreakLease(ctx, blobBreakLeaseOptions, modifiedAccessConditions)
+	return resp, handleError(err)
+}
+
+// ChangeLease changes the blob's lease ID.
+// For more information, see https://docs.microsoft.com/rest/api/storageservices/lease-blob.
+func (blc *BlobLeaseClient) ChangeLease(ctx context.Context, options *ChangeLeaseBlobOptions) (BlobChangeLeaseResponse, error) {
+	leaseId, proposedLeaseId, modifiedAccessConditions := options.pointers()
+	resp, err := blc.client.ChangeLease(ctx, leaseId, proposedLeaseId, nil, modifiedAccessConditions)
+	return resp, handleError(err)
+}
+
+// RenewLease renews the blob's previously-acquired lease.
+// For more information, see https://docs.microsoft.com/rest/api/storageservices/lease-blob.
+func (blc *BlobLeaseClient) RenewLease(ctx context.Context, options *RenewLeaseBlobOptions) (BlobRenewLeaseResponse, error) {
+	leaseId, renewLeaseBlobOptions, modifiedAccessConditions := options.pointers()
```

Comment:
I thought the point of the lease clients were to make it simpler for the end user, by remembering the lease information, and supplying it to the generated code when necessary?

Reply:
This seems to only rely on the options supplied, not the internal parameters.